solidity/random-beacon/contracts/RandomBeacon.sol

```diff
@@ -562,6 +647,59 @@ contract RandomBeacon is Ownable {
         }
     }
 
+    /// @notice Punishes the given operators by kicking them out from the
+    ///         sortition pool and blocking their reinsertion for a given period
+    ///         of time.
+    /// @dev By the way, this function releases gas deposits made by operators
+    ///      during their registration. See `registerOperator` function. This
+    ///      action makes punishments cheaper gas-wise.
+    /// @param ids IDs of punished operators.
+    /// @param punishmentDuration Duration of the punishment period in seconds.
+    ///
+    /// TODO: This method can be probably optimized in future (depends on
+    ///       changes in the sortition pool). Until then we need to make it
+    ///       that way because we need to make sure we remove operators which
+    ///       are actually in the pool to avoid revert. Two loops are needed
+    ///       because Solidity doesn't allow for memory arrays and mappings.
+    function punishOperators(uint32[] memory ids, uint256 punishmentDuration)
+        internal
+    {
+        address[] memory operators = sortitionPool.getIDOperators(ids);
+        uint256 operatorsInPoolCount = 0;
+
+        for (uint256 i = 0; i < operators.length; i++) {
+            if (sortitionPool.isOperatorInPool(operators[i])) {
```

Comment:
We are not interested about the total operators in the pool count. We are interested in which operators from our list are actually in the pool. It's important to remove only those operators to avoid reverting upon removal.

Reply:
> We are not interested about the total operators in the pool count. We are interested in which operators from our list are actually in the pool.

yes, what I meant is to pass an array of `operators` and make the sortition pool do the counting instead of calling a contract for every operator on our side. But if the plan is to get rid of `removeOperators` then I guess this comment won't be relevant.

Reply:
Yeah, I'd hold to making any changes in sortition pools until we have the changes around rewards blocking here.

Reply:
See https://github.com/keep-network/keep-core/pull/2671/commits/223619641ed245e3a41ab5213ddddf74d0fa1a33